### DISCUSSION TITLE
[Camera Support]: Frigate 0.14.0 single camera view not working on one specific camera

### DISCUSSION THREAD

**Discussion by nutsoh** [blakeblackshear/frigate#13226] [Camera Support] [answered]
### Describe the problem you are having

In the process of moving my frigate installation to a new machine in a somewhat convoluted configuration (which may well be the underlying cause here) I set up frigate:stable-rk docker image as a stack on Portainer running in a proxmox container in a Radxa Rock5B. Took a little bit but it works great with an m.2 coral and all.

I have two cameras, some older ieGeek IP Cameras that are working just great, using go2rtc for the main stream so I can restream to homeassistant.

Today I added a third camera intending only to log its power usage to size an off grid battery installation for it, and it being the same as the others, same hardware and firmware version, I assumed it would just work and it almost did; It detects and records, it shows in the live view alongside the others, all is great except when I click on the live stream to see the individual camera full screen it just keeps the spinning thing spinning, forever. This particular camera is called Spare204.

I can use VLC to open the go2rtc stream externally as rtsp:<frigate-ip>:8554/Spare204, I opened 1984 to check go2rtc and it shows no errors, and its web viewer can stream perfectly, there are no errors on frigate's logs, the only thing I get is if inspecting the <frigate-ip>:5000/#Spare204 page there's a "Uncaught (in promise) DOMException: The element has no supported sources." error in the console.

CPU hovers around 11%, inference is ~12ms, there's 2GB available to Portainer (which is only running itself and frigate), less than 1GB used, it is recording to a samba share that has multiple TBs available.

I also tried removing the other cameras, just in case this was a "third camera special", to no avail.

Like I said, this might be a product of my setup, but it feels weird that everything works great except that one camera page. Not quite sure where to go from here, for lack of any error to point me in some direction.

FYI, before copying the logs I restarted frigate, clicked on each video thumb, Spare204 last, and all worked except the latter as expected.

### Steps to reproduce

For my setup, I just click on the video thumbnail in the Live page which loads the #Spare204 anchor, and get a spinning wheel forever. Not sure if there is any way to have this be reproduced outside my specific configuration.

### Version

0.14.0-da913d8

### Frigate config file

```yaml
database:
  path: /config/frigate.db
detectors:
  coral:
    type: edgetpu
    device: pci
objects: # Object to detect - https://docs.frigate.video/configuration/objects
  track:
    - person
    - bicycle
    - car
    - motorcycle
    - bird
    - cat
    - dog
    - horse
    - sheep
    - cow

record: # how long to record before and after motion
  events:
    pre_capture: 7
    post_capture: 10
birdseye:
  enabled: true
  mode: motion
cameras:
  Gate:   # <------ Name the camera
    ffmpeg:
      inputs:
        - path: rtsp://admin:pass@<gatecamip>/12         # Update with your camera info (Normally the lower res stream, 1280*720 would be a good detection res)
          roles:
            - detect
        - path: rtsp://127.0.0.1:8554/Gate         # Update with your camera info (Normally the higher res stream, used for actual recording)
          input_args: preset-rtsp-restream
          roles:
            - record
    detect:
      width: 640       # <---- update for your camera's resolution
      height: 352       # <---- update for your camera's resolution
    record:     # what you want Frigate to record
      enabled: true
      retain:       # uncomment this section if you want Frigate to record this camera 24/7. This can use a lot of space, you have been warned!
        days: 1
        mode: all
      events:       # this is the default recording mode where events are recorded, the number of days it saves events for is 10.
        retain:
          mode: active_objects
          default: 10
    snapshots:
      enabled: true
      retain:
        default: 10
    motion:
      mask:
        - 640,21,640,0,470,0,471,26
    objects:
      filters:
        person:
          mask:
            - 346,304,468,290,450,227,363,242
            - 406,111,387,109,381,145,410,140
            - 207,175,241,159,235,136,200,139
  Coop1:   # <------ Name the camera
    ffmpeg:
      inputs:
        - path: rtsp://admin:pass@<coop1camip>/12         # Update with your camera info (Normally the lower res stream, 1280*720 would be a good detection res)
          roles:
            - detect
        - path: rtsp://admin:pass@<coop1camip>/11         # Update with your camera info (Normally the higher res stream, used for actual recording)
          roles:
            - record
    objects:
      track: []
    detect:
      width: 640       # <---- update for your camera's resolution
      height: 352       # <---- update for your camera's resolution
      enabled: false
    record:     # what you want Frigate to record
      enabled: false
      retain:       # uncomment this section if you want Frigate to record this camera 24/7. This can use a lot of space, you have been warned!
        days: 1
        mode: all
      events:       # this is the default recording mode where events are recorded, the number of days it saves events for is 10.
        retain:
          mode: active_objects
          default: 0
    snapshots:
      enabled: false
      retain:
        default: 10
  Spare204:
    ffmpeg:
      inputs:
        - path: rtsp://admin:pass@<spare204camip>/12         # Update with your camera info (Normally the lower res stream, 1280*720 would be a good detection res)
          roles:
            - detect
        - path: rtsp://127.0.0.1:8554/Spare204         # Update with your camera info (Normally the higher res stream, used for actual recording)
          input_args: preset-rtsp-restream
          roles:
            - record
    objects:
      track: []
    detect:
      enabled: true
      width: 640       # <---- update for your camera's resolution
      height: 352       # <---- update for your camera's resolution
    record:     # what you want Frigate to record
      enabled: true
      retain:       # uncomment this section if you want Frigate to record this camera 24/7. This can use a lot of space, you have been warned!
        days: 1
        mode: all
      events:       # this is the default recording mode where events are recorded, the number of days it saves events for is 10.
        retain:
          mode: active_objects
          default: 10
    snapshots:
      enabled: true
      retain:
        default: 10
go2rtc:
  streams:
    Gate:
      - rtsp://admin:pass@<gatecamip>/11
    Spare204:
      - rtsp://admin:pass@<spare204camip>/11
  webrtc:
    candidates:
      - <frigateip>:8555
version: 0.14
```

### Relevant log output

```shell
frigate:

2024-08-20 21:59:07.783834048  [INFO] Preparing Frigate...
2024-08-20 21:59:07.913553296  [INFO] Starting Frigate...
2024-08-20 21:59:14.418403182  [2024-08-20 21:59:14] frigate.app                    INFO    : Starting Frigate (0.14.0-da913d8)
2024-08-20 21:59:14.422191527  [2024-08-20 21:59:14] frigate.util.config            INFO    : Checking if frigate config needs migration...
2024-08-20 21:59:14.595151400  [2024-08-20 21:59:14] frigate.util.config            INFO    : frigate config does not need migration...
2024-08-20 21:59:15.128509793  [2024-08-20 21:59:15] frigate.util.services          WARNING : Did not detect hwaccel, using a GPU for accelerated video decoding is highly recommended
2024-08-20 21:59:15.246228483  [2024-08-20 21:59:15] peewee_migrate.logs            INFO    : Starting migrations
2024-08-20 21:59:15.246583139  [2024-08-20 21:59:15] peewee_migrate.logs            INFO    : There is nothing to migrate
2024-08-20 21:59:15.281904375  [2024-08-20 21:59:15] frigate.app                    INFO    : Recording process started: 399
2024-08-20 21:59:15.305439422  [2024-08-20 21:59:15] frigate.app                    INFO    : Recording process started: 408
2024-08-20 21:59:15.315575119  [2024-08-20 21:59:15] frigate.app                    INFO    : go2rtc process pid: 110
2024-08-20 21:59:15.402534687  [2024-08-20 21:59:15] detector.coral                 INFO    : Starting detection process: 427
2024-08-20 21:59:15.411831576  [2024-08-20 21:59:15] frigate.detectors.plugins.edgetpu_tfl INFO    : Attempting to load TPU as pci
2024-08-20 21:59:15.460179422  [2024-08-20 21:59:15] frigate.app                    INFO    : Output process started: 429
2024-08-20 21:59:15.472937708  [2024-08-20 21:59:15] frigate.detectors.plugins.edgetpu_tfl INFO    : TPU found
2024-08-20 21:59:15.489776871  [2024-08-20 21:59:15] frigate.app                    INFO    : Camera processor started for Gate: 457
2024-08-20 21:59:15.516010629  [2024-08-20 21:59:15] frigate.app                    INFO    : Camera processor started for Coop1: 463
2024-08-20 21:59:15.542320801  [2024-08-20 21:59:15] frigate.app                    INFO    : Camera processor started for Spare204: 468
2024-08-20 21:59:15.568411646  [2024-08-20 21:59:15] frigate.app                    INFO    : Capture process started for Gate: 471
2024-08-20 21:59:15.595903324  [2024-08-20 21:59:15] frigate.app                    INFO    : Capture process started for Coop1: 479
2024-08-20 21:59:15.628162444  [2024-08-20 21:59:15] frigate.app                    INFO    : Capture process started for Spare204: 488

----

go2rtc:

2024-08-20 21:59:07.874268428  [INFO] Preparing new go2rtc config...
2024-08-20 21:59:09.233859378  [INFO] Not injecting WebRTC candidates into go2rtc config as it has been set manually
2024-08-20 21:59:09.623167910  [INFO] Starting go2rtc...
2024-08-20 21:59:09.801914568  21:59:09.801 INF go2rtc platform=linux/arm64 revision=b2399f3 version=1.9.2
2024-08-20 21:59:09.801987191  21:59:09.801 INF config path=/dev/shm/go2rtc.yaml
2024-08-20 21:59:09.803275444  21:59:09.803 INF [api] listen addr=:1984
2024-08-20 21:59:09.803285360  21:59:09.803 INF [rtsp] listen addr=:8554
2024-08-20 21:59:09.804382577  21:59:09.804 INF [webrtc] listen addr=:8555
2024-08-20 21:59:17.781594128  [INFO] Starting go2rtc healthcheck service...
```

### Operating system

Proxmox

### Install method

Docker Compose

### Network connection

Wired

### Camera make and model

ieGeek 2.0M Pixels IP Camera

**Comment by hawkeye217** [comment, marked as answer]
There could be several reasons for this. Can you try the latest dev build of Frigate and see if that corrects the issue for you? I recently committed some fixes for Live view.

**Comment by nutsoh** [comment]
That was quick! I just tried master-22fe261-rk, and unfortunately that didn't change anything. Although I have been trying to keep the cameras configs as in sync as possible, given two work and the other does not, I did run some extra tests just before this version update with varying iframes, lowering the bitrate and fps, turning off audio which I'm not using but is on by default in all 3 cameras, no change. I also forgot to mention originally that I'm using the Brave browser on windows 11, but I tested with Edge with consistent results.

This is not a priority as I haven't seen it reported anywhere. If you need me to do anything to help debug this I'm available, but honestly we can just shrug this off as my camera misbehaving somehow. The only weird thing is all the streams seem to be working great, it records from the main one just fine, and I can access it externally as mentioned. As weird as it gets...

**Reply by hawkeye217** [reply]
Give `dev-d96f76c-rk` a try.

**Reply by nutsoh** [reply]
You did say "latest dev build", sorry. Bad habits from work, master branch gets used for development. 

And it does work! There are a few caveats and details I'd like to share, though. I didn't time it before, but the go2rtc proxied cameras (all but Coop1) take 10>20 seconds to load, though they are not lagging by anything close to that once loaded, rather a very comfortable 1>2 secs. I'm pretty sure it was a lot faster before for the Gate camera, and Coop1, which is direct rtsp withough go2rtc loads in 3>4 seconds.
CPU is higher now too, ~17%, but that means very little without having previously logged where the CPU was being used to compare to now. 

I tried Chrome on my Android 13 and with the previous (master-*-rk) version none of the go2rtc cameras was loading, and now both work fine, bar the long warmup.

When I say "camera loading" I mean on the single camera page, the live thumbnails have never failed nor lagged.

So, I'm calling this resolved unless you want me to test anything else, feel free to close this discussion if you feel so inclined and thanks for the amazingly quick help!

**Reply by hawkeye217** [reply]
Great! Happy to help.

There are a few factors that may contribute to a delay in bringing up a live stream in single cam view. Your camera sends keyframes at a specific interval, and streaming can't begin until a certain number of keyframes have been received. Many times this is an adjustable parameter in your camera's firmware (sometimes called the i-frame interval or interframe space). Matching the i-frame interval to your camera's frame rate is the recommendation. See this page for more on keyframes: https://gardinal.net/understanding-the-keyframe-interval/

Another reason why it may take a while is that your browser is not able to load the high-res stream because the audio or video codec is unsupported or because the stream can't be decoded for some reason. This may be what you experienced when you saw the `DOMException: The element has no supported sources.` error. Frigate will try buffering your high-res stream for 3 seconds, and if a keyframe or sufficient data haven't been received, it will fall back to the WebRTC player (if you have WebRTC set up) and then if that isn't configured, it'll finally fall back to the basic jsmpeg player. This also then takes a little time to load.

Browser support for various codecs is a little crazy due to copyrights/patents. Check out this chart in go2rtc's readme: https://github.com/AlexxIT/go2rtc?tab=readme-ov-file#codecs-madness

So it's hard to say exactly what you are experiencing, but at least you have a working live view now.

I'll close this discussion but feel free to comment more or reopen it if you have any additional issues.
